tests/openshift/templates/bddstack.bc.yaml

```diff
@@ -3,123 +3,169 @@
+          rpm -V $INSTALL_PKGS && \
+          yum clean all -y
+
+      # Install Gradle
+
+      RUN mkdir /opt/gradle && \
+          wget https://services.gradle.org/distributions/gradle-4.2.1-bin.zip -P /opt/gradle && \
+          unzip /opt/gradle/gradle-4.2.1-bin.zip -d /opt/gradle && \
+          rm /opt/gradle/gradle-4.2.1-bin.zip
+
+      RUN if [ ! -d /home/jenkins/.pki ] ; then mkdir /home/jenkins/.pki; fi &&
+      \
+          chmod 777 /home/jenkins/.pki
+
+      # In order to drop the root user, we have to make some directories world
+
+      # writable as OpenShift default security model is to run the container
+      under
+
+      # random UID.
+
+      RUN mkdir -p /opt/app-root  && \
+          chown -R 1001:0 /opt/app-root && \
+          chmod -R og+rwx /opt/app-root
+
+      USER 1001
+    type: Dockerfile
+  strategy:
+    dockerStrategy:
+      env:
+        - name: OPENSHIFT_JENKINS_JVM_ARCH
+          value: x86_64
+        - name: BASEURL
+          valueFrom:
+            configMapKeyRef:
+              key: BASEURL
+              name: test-config
+        - name: BROWSERSTACK_VERSION
+          valueFrom:
+            configMapKeyRef:
+              key: BROWSERSTACK_VERSION
+              name: test-config
+        - name: CHROMEDRIVER_VERSION
+          valueFrom:
+            configMapKeyRef:
+              key: CHROMEDRIVER_VERSION
+              name: test-config
+        - name: GEB_VERSION
+          valueFrom:
+            configMapKeyRef:
+              key: GEB_VERSION
+              name: test-config
+        - name: GECKODRIVER_VERSION
+          valueFrom:
+            configMapKeyRef:
+              key: GECKODRIVER_VERSION
+              name: test-config
+        - name: GROOVY_VERSION
+          valueFrom:
+            configMapKeyRef:
+              key: GROOVY_VERSION
+              name: test-config
+        - name: HTTPBUILDER_VERSION
+          valueFrom:
+            configMapKeyRef:
+              key: HTTPBUILDER_VERSION
+              name: test-config
+        - name: MYSQLCONNECTORJAVA_VERSION
+          valueFrom:
+            configMapKeyRef:
+              key: MYSQLCONNECTORJAVA_VERSION
+              name: test-config
+        - name: PROJECT_NAME
+          valueFrom:
+            configMapKeyRef:
+              key: PROJECT_NAME
+              name: test-config
+        - name: SELENIUM_VERSION
+          valueFrom:
+            configMapKeyRef:
+              key: SELENIUM_VERSION
+              name: test-config
+        - name: SLF4J_VERSION
+          valueFrom:
+            configMapKeyRef:
+              key: SLF4J_VERSION
+              name: test-config
+        - name: SPOCKREPORT_VERSION
+          valueFrom:
+            configMapKeyRef:
+              key: SPOCKREPORT_VERSION
+              name: test-config
+        - name: SPOCK_VERSION
+          valueFrom:
+            configMapKeyRef:
+              key: SPOCK_VERSION
+              name: test-config
```

Comment:
A lot of this can be condensed and simplified down to importing an entire configmap manifest object. When you di it this way, the key value becomes the environment variable that will get inserted into the Pod on instantiation (reduces redundancy and potential typo errors, but onus now sits heavily on configmap instead):
```yaml
              envFrom:
                - configMapRef:
                    name: test-config
```

Reply:
So where exactly would you put this? Can you show me in the file?